src/modules/Dashboard/features/Respondents/Popups/DataExportPopup/DataExportPopup.hooks.tsx

```diff
@@ -0,0 +1,146 @@
+
+  const distributeTasks = useCallback(() => {
+    workersRef.current.forEach((worker) => {
+      if (worker.isIdle && taskQueueRef.current.length > 0) {
+        const currentTask = taskQueueRef.current.shift();
+        if (!currentTask) return;
+
+        const { pageData, page } = currentTask;
+        worker.isIdle = false;
+        worker.postMessage({
+          encryptedData: pageData,
+          encryptionInfoFromServer,
+          page,
+          privateKey,
+          hasSuffix: limitRef.current > 1,
+          filters,
+          shouldLogDataInDebugMode,
+          ItemResponseType,
+        });
+
+        setCurrentPage((prevState) => prevState + 1);
+      }
+    });
+  }, [encryptionInfoFromServer, privateKey, filters, shouldLogDataInDebugMode]);
+
+  const processData = useCallback(
+    (pageData: ExportDataResult, page: number) => {
+      if (workersRef.current.length === 0 || processedPagesRef.current.has(page)) return;
+
+      taskQueueRef.current.push({ pageData, page });
+      distributeTasks();
+    },
+    [distributeTasks],
+  );
+
+  const handleAllTasksCompleted = () => {
+    // check if all pages are processed and all workers are idle
+    const allPagesProcessed =
+      processedPagesRef.current.size === limitRef.current &&
+      finishedPagesRef.current.size === limitRef.current;
+    const allWorkersIdle = workersRef.current.every((worker) => worker.isIdle);
+
+    if (!dataExportingApiRef.current && allPagesProcessed && allWorkersIdle) {
+      setDataIsExporting(false);
+      handleExportPopupClose();
+      Mixpanel.track('Export Data Successful', {
+        'Applet ID': appletId,
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (!exportDataQueue) return;
+
+    const processingBatch = exportDataQueue[currentPage - 1];
+
+    if (processingBatch) {
+      processData(processingBatch, currentPage);
+    }
+  }, [exportDataQueue, currentPage, processData]);
+
+  useEffect(() => {
+    const workers: IdleWorker[] = Array.from({ length: NUM_WORKERS }, () => {
+      const worker = new DecryptionWorker() as IdleWorker;
+      worker.isIdle = true;
+      worker.onmessage = async (event: WorkerOnMessageEvent) => {
+        const { success, decryptedData, error, page, hasSuffix, filters } = event.data;
+        worker.isIdle = true;
+        processedPagesRef.current.add(page);
+
+        if (success) {
+          await exportDecryptedDataSucceed({
+            suffix: hasSuffix ? getExportDataSuffix(page) : '',
+            filters,
+          })(decryptedData);
+
+          finishedPagesRef.current.add(page);
+          handleAllTasksCompleted();
+        } else {
+          console.error('Decryption failed:', error);
+        }
+
+        distributeTasks();
+      };
+
+      return worker;
+    });
+
+    workersRef.current = workers;
+
+    return () => {
+      workers.forEach((worker) => worker.terminate());
+    };
+    // create workers only once when the component mounts
```

Comment:
Please elaborate on why they shouldn't be recreated. AFAIK, React's Strict Mode runs every effect twice in dev. So, if nothing happens in dev mode then there shouldn't; be a problem to recreated those workers

Reply:
Why Workers Shouldn't Be Recreated:

- Resource Efficiency: Creating web workers is a resource-intensive operation. Recreating them unnecessarily would consume more CPU and memory, leading to performance issues.

- Task Continuity: If workers were recreated during the component's lifecycle, any ongoing tasks assigned to the previous workers would be lost, potentially leading to incomplete processing or errors.

- React Strict Mode: In development mode, React's Strict Mode intentionally mounts, unmounts, and remounts components to help identify side effects and potential issues. Properly managed workers that function correctly during this process indicate that the code is handling side effects properly. However, recreating workers every time the component re-renders or runs effects can lead to unintended consequences, such as multiple workers processing the same tasks.

- Avoiding Duplicated Workers: Recreating workers on every render or effect run could lead to multiple instances of workers processing the same data or conflicting with each other, resulting in bugs and unpredictable behavior.